### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by vanityx**

### CODE REVIEW

**Comment by almeister** [comment]
Hey Sasha, this looks good!  You’ve used the kv file well to refactor the code.  I’m going to leave some specific feedback soon that you can use to refine the code.

Code Reviews and Pull Requests in general are used to double-check your code for errors or improvements before you merge code into the master branch.  After it’s merged, then the extra branch can be safely deleted.  I’ll show you this tomorrow.

File: source/main.py [hunk 1]
```diff
@@ -3,133 +3,39 @@
...
-        self.btn2.bind(text=self.change_nose)
-        self.btn3.bind(text=self.change_lips)
-
-    def change_base(self, spinner, text):
-        length = len(text)
-        last_num = str(text[length - 1])
-        curr_nose = self.nose_img.source[14]
-        self.base_img.source = '../assets/base' + last_num + '.png'
+    def change_base(self, value):
+        length = len(value)
```
Comment (almeister) [inline source/main.py:21]: Looks like length and last_num are redundant.  Remove them.

Comment (almeister) [reply source/main.py:21]: Oh, I see - you’re using last_num to build your image source string.

File: source/main.py [hunk 2]
```diff
@@ -3,133 +3,39 @@
...
+        last_num = str(value[length - 1])
+        curr_nose = self.ids.nose_img.source[14]
```
Comment (almeister) [inline source/main.py:23]: The number 14 is a magic number (Google this).  I’m guessing you’re extracting the number here to name the image source.

Could you use the button id number instead?

File: source/main.py [hunk 3]
```diff
@@ -3,133 +3,39 @@
...
-    def change_lips(self, spinner, text):
-        if "1" in text:
-            self.lips_img.source = '../assets/lips1.png'
-        if "2" in text:
-            self.lips_img.source = '../assets/lips2.png'
-        if "3" in text:
-            self.lips_img.source = '../assets/lips3.png'
-        if "4" in text:
-            self.lips_img.source = '../assets/lips4.png'
+    def change_lips(self, text):
```
Comment (almeister) [inline source/main.py:35]: Hmm, I’ll need to check this on a computer (I’m using my phone).

It looks like the text here should just be “LIPS”.  In which case, will it ever find a number and assign the image source as intended?

Am I missing something?

File: source/main.py (hunk 2 above)
Comment (almeister) [reply source/main.py:23]: If not, you can replace 14 with (length of that string - 1), assuming the number is the last character.

Comment (almeister) [reply source/main.py:23]: OK, I see what you’re doing with the numbering - selecting type and colour.  Ignore my comments.

File: source/main.py (hunk 3 above)
Comment (vanityx) [reply source/main.py:35]: Nope! The text for lips always has a number.

Comment (vanityx) [reply source/main.py:35]: I changed 'text' to 'value' so that this is clearer.

**Pull request merged**